src/main/java/Task.java

```diff
@@ -0,0 +1,33 @@
+public class Task {
```

Comment:
I like how you use inheritance to reduce code duplication! Since ```Task``` does not seem to be instantiated in your code, maybe you could make it an abstract class!